src/3rdparty/squirrel/squirrel/sqvm.cpp

```diff
@@ -1284,9 +1284,10 @@ bool SQVM::FallBackGet(const SQObjectPtr &self,const SQObjectPtr &key,SQObjectPt
 	case OT_STRING:
 		if(sq_isnumeric(key)){
 			SQInteger n=tointeger(key);
-			if(abs((int)n)<_string(self)->_len){
-				if(n<0)n=_string(self)->_len-n;
-				dest=SQInteger(_stringval(self)[n]);
+			std::string_view str = _stringval(self);
+			if(std::abs(n) < static_cast<SQInteger>(str.size())){
+				if(n<0)n=str.size()-n;
```

Comment:
How did this never explode?
```suggestion
				if(n<0)n=str.size()+n;
```

Reply:
Probably because nobody attempted to use negative indices in strings yet.